Octokit/Clients/AuthorizationsClient.cs

```diff
@@ -95,18 +181,11 @@ public Task<Authorization> Get(int id)
                 note_url = newAuthorization.NoteUrl
             };
 
-            if (String.IsNullOrWhiteSpace(newAuthorization.Fingerprint))
-            {
-                // use classic API
-                var endpoint = ApiUrls.AuthorizationsForClient(clientId);
-                return ApiConnection.Put<ApplicationAuthorization>(endpoint, requestData);
-            }
-            else
-            {
-                // use new API
-                var endpoint = ApiUrls.AuthorizationsForClient(clientId, newAuthorization.Fingerprint);
-                return ApiConnection.Put<ApplicationAuthorization>(endpoint, requestData, null, previewAcceptsHeader);
-            }
+            var endpoint = string.IsNullOrWhiteSpace(newAuthorization.Fingerprint)
+                ? ApiUrls.AuthorizationsForClient(clientId)
+                : ApiUrls.AuthorizationsForClient(clientId, newAuthorization.Fingerprint);
+
+            return ApiConnection.Post<ApplicationAuthorization>(endpoint, requestData);
```

Comment:
I *really really really* think this should be a `Put` as per https://developer.github.com/v3/oauth_authorizations/#get-or-create-an-authorization-for-a-specific-app